cdxev/amend/operations.py

```diff
@@ -119,53 +119,62 @@ class InferSupplier(Operation):
+
     For all of the URLs there is the additional condition that they must utilize the http or https
     scheme.
     """
 
     def infer_supplier(self, component: dict) -> None:
-        if "supplier" in component:
-            return
-
-        if "externalReferences" in component:
-            accepted_references = ("website", "issue-tracker", "vcs")
-            accepted_url_schemes = ("http://", "https://")
-            for key in accepted_references:
-                ext_ref = next(
-                    (
-                        x
-                        for x in component["externalReferences"]
-                        if x.get("type") == key
-                    ),
-                    None,
-                )
-                if ext_ref is not None and (
-                    any(
-                        ext_ref["url"].startswith(scheme)
-                        for scheme in accepted_url_schemes
-                    )
-                ):
-                    component["supplier"] = {"url": [ext_ref["url"]]}
-                    logger.debug(
-                        "Set supplier of %s to URL: %s",
-                        component.get("bom-ref", "<no bom-ref>"),
-                        ext_ref["url"],
+        if "url" not in component.get("supplier", {}):
+
+            if "externalReferences" in component:
+                accepted_references = ("website", "issue-tracker", "vcs")
+                accepted_url_schemes = ("http://", "https://")
+                for key in accepted_references:
+                    ext_ref = next(
+                        (
+                            x
+                            for x in component["externalReferences"]
+                            if x.get("type") == key
+                        ),
+                        None,
                     )
-                    break
-
-        if "publisher" in component:
-            component["supplier"] = {"name": component["publisher"]}
-            return
-
-        if "author" in component:
-            component["supplier"] = {"name": component["author"]}
-            return
+                    if ext_ref is not None and (
+                        any(
+                            ext_ref["url"].startswith(scheme)
+                            for scheme in accepted_url_schemes
+                        )
+                    ):
+                        component["supplier"] = component.get("supplier", {})
+                        component["supplier"]["url"] = [ext_ref["url"]]
+                        logger.debug(
+                            "Set supplier of %s to URL: %s",
+                            component.get("bom-ref", "<no bom-ref>"),
+                            ext_ref["url"],
+                        )
+                        break
+
+        if "name" not in component.get("supplier", {}):
+
+            if "publisher" in component:
+                component["supplier"] = component.get("supplier", {})
```

Comment:
maybe add this line before the two `if`s?

Reply:
Then this might result in an empty 'supplier' getting created.